_includes/templates/install/ubuntu-queue-kafka.md

```diff
@@ -0,0 +1,112 @@
+#### Kafka Installation
+
+[Apache Kafka](https://kafka.apache.org/) is an open-source stream-processing software platform. 
+ThingsBoard uses Kafka like queue service for some API calls between micro-services.
+
+##### Install ZooKeeper
+
+Kafka uses [ZooKeeper](https://zookeeper.apache.org/) so you need to first install ZooKeeper server:
+
+```text
+sudo apt-get install zookeeper
+```
+{: .copy-code}
+
+##### Install Kafka
+
+```text
+wget http://www-us.apache.org/dist/kafka/2.4.0/kafka_2.13-2.4.0.tgz
+
+tar xzf kafka_2.13-2.4.0.tgz
+
+sudo mv kafka_2.13-2.4.0 /usr/local/kafka
+```
+
+##### Setup ZooKeeper Systemd Unit file
+
+Create systemd unit file for Zookeeper:
+```text
+sudo nano /etc/systemd/system/zookeeper.service
+```
+{: .copy-code}
+
+Add below contnet:
+```bash
+[Unit]
+Description=Apache Zookeeper server
+Documentation=http://zookeeper.apache.org
+Requires=network.target remote-fs.target
+After=network.target remote-fs.target
+
+[Service]
+Type=simple
+ExecStart=/usr/local/kafka/bin/zookeeper-server-start.sh /usr/local/kafka/config/zookeeper.properties
+ExecStop=/usr/local/kafka/bin/zookeeper-server-stop.sh
+Restart=on-abnormal
+
+[Install]
+WantedBy=multi-user.target
+```
+{: .copy-code}
+
+##### Setup Kafka Systemd Unit file
+
+Create systemd unit file for Kafka:
+```text
+sudo nano /etc/systemd/system/kafka.service
+```
+{: .copy-code}
+
+Add the below content. Make sure **to replace** "PUT_YOUR_JAVA_PATH" with your **real JAVA_HOME path** as per the Java installed on your system: 
+```bash
+[Unit]
+Description=Apache Kafka Server
+Documentation=http://kafka.apache.org/documentation.html
+Requires=zookeeper.service
+
+[Service]
+Type=simple
+Environment="JAVA_HOME=/usr/lib/jvm/PUT_YOUR_JAVA_PATH"
```

Comment:
Please, ensure that you replace "JAVA_HOME_PATH"  with your **real JAVA_HOME path**. By default it should be located here:
/usr/lib/jvm/

Reply:
done